src/collectors/kafkastat/test/testkafka.py

```diff
@@ -108,10 +108,10 @@ def test_query_mbean(self, get_mock):
         get_mock.return_value = self._get_xml_fixture('mbean.xml')
 
         expected_metrics = {
-            'kafka.logs.mytopic-1.CurrentOffset': long('213500615'),
-            'kafka.logs.mytopic-1.NumAppendedMessages': long('224634137'),
-            'kafka.logs.mytopic-1.NumberOfSegments': int('94'),
-            'kafka.logs.mytopic-1.Size': long('50143615339'),
+            'kafka.kafka.logs.mytopic-1.CurrentOffset': long('213500615'),
```

Comment:
given the lines:
`key_prefix = prefix + "." + keys.pop(0).split('=')[-1]`
and
`keys = map(lambda i: i.replace('=', '_').replace('.', '_'), keys)`

Consider this key:
`kafka.server:type=BrokerTopicMetrics,name=BytesRejectedPerSec,topic=escher_heartbeat_event`

It sounds like we would end up with:

`kafka.kafka.server.type_BrokerTopicMetrics.name_BytesRejectedPerSec.topic_escher_heartbeat_event.Count`

is this correct?
If so then this is not preferrable, first we already start with kafka due to the path variable at the top. So we should use only the part of the prefix that we require to give needed separation (server vs kafka.server) as kafka is now redundant I think. Also I would rather remove artifacts like 'type_', 'name_', and 'topic_' as they are not needed and make your graphite paths unnecessarily long.

I supplied another patch for this here:
https://github.com/python-diamond/Diamond/issues/225

 I do like your patch as the code is more elegant than mine (non-python coder here), but would prefer to lose the parts mentioned above.